src/include/OpenImageIO/fmath.h

```diff
@@ -228,13 +228,20 @@ clamp (simd::float4 a, simd::float4 low, simd::float4 high)
 /// Multiply and add: (a * b) + c
 template <typename T>
 inline T madd (const T& a, const T& b, const T& c) {
-    // NOTE:  in the future we may want to explicitly ask for a fused
-    // multiply-add in a specialized version for float.
-    // NOTE2: GCC/ICC will turn this (for float) into a FMA unless
-    // explicitly asked not to, clang seems to leave the code alone.
+    // NOTE: GCC/ICC will turn this (for float) into a FMA unless
+    // explicitly asked not to, clang will do so if -ffp-contract=fast.
     return a * b + c;
 }
 
+#if OIIO_SIMD_SSE
+template <>
+inline simd::float4 madd (const simd::float4& a, const simd::float4& b,
+                          const simd::float4& c) {
+   // Directly access the underlying __m128 to ensure a true fma
+   return a.simd() * b.simd() + c.simd();
```

Comment:
As far as I can tell, there is no preprocessor symbol that lets me know when it has been enabled.
So at the source code level, I don't know when it's safe to use the intrinsic.